pkg/status/status.go

```diff
@@ -191,12 +259,90 @@ func (s *status) setStatusCondition(condition configv1.ClusterStatusConditionTyp
 		Message:            failingMessage,
 		LastTransitionTime: time,
 	})
+	return updatedCondition
 }
 
-// updateStatus makes the API call to update the ClusterOperator
-func (s *status) updateStatus() {
-	_, err := s.configClient.ClusterOperators().UpdateStatus(s.clusterOperator)
-	if err != nil {
-		log.Fatalf("Error %v updating ClusterOperator", err)
+// updateStatus makes the API call to update the ClusterOperator if the status has changed.
+func (s *status) updateStatus(previousStatus *configv1.ClusterOperatorStatus, updatedCondition string, message string) {
+	if reflect.DeepEqual(previousStatus, &s.clusterOperator.Status) {
```

Comment:
Are you sure you cannot leverage the library-go [helpers](https://github.com/openshift/library-go/blob/master/pkg/config/clusteroperator/v1helpers/status.go) for ClusterOperator here?